src/client/datascience/notebook/notebookControllerManager.ts

```diff
@@ -697,7 +717,6 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
                     // We are disposing as documents are closed, but do this as well
                     this.disposables.push(controller);
                     this.registeredControllers.set(controller.id, controller);
-                    this.allKernelConnections.add(kernelConnection);
```

Comment:
Removed this property, pointless maintaining state in two properties (buggy & unnecessary)